### PR TITLE
Adding service-account.yaml to kustomization.yaml

### PR DESCRIPTION
**Pull request by amolsr**
Issue #1451

Description of changes:
Adding the Service-account.yaml in kustomization.yaml to get picked up by kustomize installation.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by A-Hilaly — APPROVED** [review]
Thanks @amolsr !
/lgtm

**Comment by amolsr** [comment]
I think there is some missing confg file in sns-controller/config/crd/common/bases that is why the test are failing. @jaypipes would be needing to update the code-generator for sns for that in place. can we consider this as hotfix and merge.?

File: config/rbac/kustomization.yaml [hunk 1]
```diff
@@ -3,3 +3,4 @@ resources:
 - cluster-role-controller.yaml
 - role-reader.yaml
 - role-writer.yaml
+- service-account.yaml
```
Comment (RedbackThomson) [inline config/rbac/kustomization.yaml:6]: This is included when you generate the controller. I think the issue here isn't that it's missing, I think the project just hasn't been generated in a while and doesn't have the latest changes?

https://github.com/aws-controllers-k8s/code-generator/blob/main/templates/config/rbac/kustomization.yaml.tpl

**Comment by RedbackThomson** [comment]
/hold

This pull request needs to be generated by properly using the code-generator

**Comment by amolsr** [comment]
image
I think the code-generator is using code which is not present in code-generator repo. I saw a similar error which state /f/Projects/code-generator/scripts/../../sns-controller is not a directory.

**Comment by jaypipes** [comment]
merging this manually for now. see earlier comments for why I need to radically fix up the codegen to deal with attribute-based APIs like SNS...

**Pull request merged**